modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -50,46 +64,91 @@
     @Override
     public void validate(ExtensionModel extensionModel) throws IllegalModelDefinitionException
     {
+        MetadataTypeVisitor visitor = new MetadataTypeVisitor()
+        {
+            private Set<Class<?>> visitedClasses = new HashSet<>();
+            private ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
+
+            @Override
+            public void visitDictionary(DictionaryType dictionaryType)
+            {
+                dictionaryType.getKeyType().accept(this);
+                dictionaryType.getValueType().accept(this);
+            }
+
+            @Override
+            public void visitArrayType(ArrayType arrayType)
+            {
+                arrayType.getType().accept(this);
+            }
+
+            @Override
+            public void visitObject(ObjectType objectType)
+            {
+                Class<?> type = getType(objectType);
+
+                if (visitedClasses.add(type) && IntrospectionUtils.isInstantiable(type) && !getExposedFields(type).isEmpty())
```

Comment:
We should discuss about this logic, because it is the same one used in the SchemaBuilder for deciding if a pojo child element should be generated.